Generate PlatformManifest and package conflict overrides for Microsoft.AspNetCore.App

natemcmaster opened the pull request:
This generates additional metadata for the .NET Core SDK to resolve conflicts between Microsoft.AspNetCore.App and PackageReferences which contain the same assemblies.

Resolves https://github.com/aspnet/AspNetCore/issues/3609

Examples of the build output created:

[Microsoft.AspNetCore.App.props](https://github.com/aspnet/AspNetCore/files/2590924/Microsoft.AspNetCore.App.props.txt)
[Microsoft.AspNetCore.App.PlatformManifest.txt](https://github.com/aspnet/AspNetCore/files/2590920/Microsoft.AspNetCore.App.PlatformManifest.txt)

dsplaisted reviewed (approved):
I didn't look at the implementation, but the build output looks correct.

Commit:
Generate PlatformManifest and package conflict overrides for Microsoft.AspNetCore.App

This generates additional metadata for the .NET Core SDK to resolve conflicts between Microsoft.AspNetCore.App and PackageReferences which contain the same assemblies.

Eilon commented on .azure/pipelines/fast-pr-validation.yml:
```diff
@@ -1,7 +1,11 @@
+# Don't run CI for this config
+trigger: none
```
What's this?

natemcmaster replied:
This disables the PR validation trigger when we push to branches. PR validation only needs to run on PRs.

Eilon commented on src/Framework/test/SharedFxTests.cs:
```diff
@@ -1,29 +1,94 @@
+        {
+            var platformManifestPath = Path.Combine(TestData.GetManifestOutputDir(), "Microsoft.AspNetCore.App.PlatformManifest.txt");
+            var expectedAssemblies = TestData.GetSharedFxDependencies()
+                .Split(';', StringSplitOptions.RemoveEmptyEntries)
```
Is `RemoveEmptyEntries` really needed? Or would it just mask potential bugs where we put blank entries in the metadata?

natemcmaster replied:
Not in this case. I'm only using remove empty entries on TestData.GetSharedFxDependencies(), which is test data generated in the unit test csproj file. The data isn't used in the product.

Eilon commented on src/Framework/test/SharedFxTests.cs:
```diff
@@ -1,29 +1,94 @@
+                .Select(i =>
+                {
+                    var fileName = i.Split('|')[0];
+                    return fileName.EndsWith(".dll")
```
Pass in string comparer. This can fail on other cultures.

Eilon reviewed (approved):
A few small comments/questions.

Commit:
PR feedback

Commit:
fixup! PR feedback
